packages/network/src/v2/mode/getBlockNumberFromModeTable.ts

```diff
@@ -0,0 +1,10 @@
+import { GenericTable } from "@latticexyz/services/protobuf/ts/mode/mode";
+import { SchemaType } from "@latticexyz/schema-type";
+import { decodeValue } from "../schemas/decodeValue";
+
+export function getBlockNumberFromModeTable(table: GenericTable): number {
+  // TODO: figure out how to get correct return type from `decodeValue`
+  // TODO: downcasting this to number, because that's what sync worker, this may result in data loss for large block numbers
```

Comment:
Assuming a 1s blocktime and starting from block 0, this can support blockchains for up to
```
Math.floor(Number.MAX_SAFE_INTEGER / (60 * 60 * 24 * 365))
> 285616414
```
years 🙈 

(This ignores sub 1s chains like "automine" ofc, but should be fine for now)